sdk-docs/tutorials/flask.md

```diff
@@ -0,0 +1,491 @@
+      Valid options are: "ara", "eng", "fra", "hin", "por", "spa", "swa", and "zho".
+    source: default
+    type: string
+    value: eng
+```
+
+Now we are able to change the language setting of the `sample-flask` Flask
+application.
+
+```bash
+# Change the 'language' config to 'spa':
+ubuntu@charm-dev-vm:~/sample-flask/charm$ juju config sample-flask language=spa
+
+# Get the latest unit IP address after refresh using the juju status command
+ubuntu@charm-dev-vm:~/sample-flask/charm$ juju status
+Model        Controller  Cloud/Region        Version  SLA          Timestamp
+welcome-k8s  microk8s    microk8s/localhost  3.1.7    unsupported  06:25:08Z
+
+App           Version  Status  Scale  Charm         Channel  Rev  Address         Exposed  Message
+sample-flask           active      1  sample-flask             1  10.152.183.175  no
+
+Unit             Workload  Agent  Address      Ports  Message
+sample-flask/0*  active    idle   10.1.30.207
+
+# Inspect the new greeting message
+ubuntu@charm-dev-vm:~/sample-flask/charm$ curl 10.1.30.207:8000
+<h1 dir="auto">Hola</h1>
+```
+
+## Enable `juju integrate sample-flask mysql-k8s`
+
+Let's enhance the sample-flask Flask application by adding a new feature:
+basic visitor analytics. To implement this feature, it's necessary to connect
+the sample-flask Flask application to a database. The Juju ecosystem offers a
+range of high quality database charms which can be seamlessly integrated with
+app charms.
+
+First, checkout the `feat-database` branch of the `sample-flask` repository and
+rebuild the rock image.
+
+```bash
+ubuntu@charm-dev-vm:~/sample-flask$ git checkout feat-database
+Branch 'feat-database' set up to track remote branch 'feat-database' from 'origin'.
+Switched to a new branch 'feat-database'
+
+# Change the version of rock in the rockcraft.yaml file again, repack the rock image
+ubuntu@charm-dev-vm:~/sample-flask$ sed -i "s/version: '0.2'/version: '0.3'/g" rockcraft.yaml
+ubuntu@charm-dev-vm:~/sample-flask$ rockcraft pack
+deleting current features configuration
+deleting current features configuration
+Packed sample-flask_0.3_amd64.rock
+
+# Push the rock image to the microk8s container registry, use the tag 'sample-flask:database' this time
+ubuntu@charm-dev-vm:~/sample-flask$ /snap/rockcraft/current/bin/skopeo --insecure-policy copy --dest-tls-verify=false oci-archive:sample-flask_0.3_amd64.rock docker://localhost:32000/sample-flask:database
+Getting image source signatures
+Copying blob 29202e855b20 skipped: already exists
+Copying blob b6eb7364648a done
+Copying blob f0c09d32dd6a done
+Copying blob bb9e380ad050 done
+Copying blob a151c1ce9737 done
+Copying config 4a65989048 done
+Writing manifest to image destination
+Storing signatures
+```
+
+Then, in your `charm/charmcraft.yaml` file, add the declaration for requiring
+the connectivity to a postgresql database charm.
+
+```yaml
+requires:
+  postgresql:
+    interface: postgresql_client
+    limit: 1
+```
```

Comment:
Cool that the charmer doesn't need to worry about events etc! I think we should still include notes signalling what the general logic is and how the extension takes care of most of it but the user can always deep-dive if they want to customize and optimize.